nixos/tests/vaultwarden-litestream: init

ak2k opened the pull request:
## Description

Integration test for Vaultwarden with Litestream continuous replication.
Validates cross-service file access and disaster recovery.

> **Note:** This PR depends on #500605 (litestream 0.3.13 → 0.5.10). The first 3 commits are from that PR — **only the last commit is for review here.** Once #500605 merges, this can be rebased to a single commit.

## Test coverage

- **Permission setup**: litestream user in vaultwarden group, chmod via ExecStartPre
- **Backup verification**: `sync -wait` via IPC socket for deterministic replication, then restore and verify test data
- **Disaster recovery**: stop services, delete DB, restore from backup, verify checksum match
- **Service recovery**: restart both services, confirm data persists (duplicate registration returns 400)

## Things done

- [x] Uses `runTest` module style (#386873)
- [x] Added myself to maintainer-list.nix
- [x] Test passes (NixOS VM test, aarch64-linux) — verified on nixos host
- [x] Uses litestream 0.5.x IPC socket (`sync -wait`) for deterministic sync verification
- [x] `nix build .#nixosTests.vaultwarden-litestream` — pass
- [x] `nix build .#nixosTests.litestream` — pass (no regression)

## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review rev 68f8d064842d`

<details>
<summary>Results for <code>aarch64-linux</code></summary>

```
1 package updated:
litestream (0.3.13 → 0.5.10)

1 package built:
litestream
```

</details>

<details>
<summary>NixOS VM test results</summary>

```
$ nix build .#nixosTests.vaultwarden-litestream  # pass (298s)
$ nix build .#nixosTests.litestream               # pass (no regression)
```

</details>

---

cc @dotlambda @SuperSandro2000 (vaultwarden maintainers)
cc @fbrs (litestream maintainer)

acid-bong commented:
Format your edits with nixfmt

dotlambda reviewed (commented):
I highly suspect you used an LLM and don't appreciate the extra work that creates.

dotlambda commented:
> * This integration pattern is valuable documentation for users deploying Vaultwarden with backup solutions

Why do we need a test and not just a paragraph on wiki.nixos.org?
Did this break in the past?

SuperSandro2000 commented on nixos/tests/all-tests.nix:
```diff
@@ -1543,6 +1543,7 @@ in
   vault-dev = runTest ./vault-dev.nix;
   vault-postgresql = runTest ./vault-postgresql.nix;
   vaultwarden = discoverTests (import ./vaultwarden.nix);
+  vaultwarden-litestream = handleTest ./vaultwarden-litestream.nix { };
```
https://github.com/NixOS/nixpkgs/issues/386873

SuperSandro2000 commented on nixos/tests/vaultwarden-litestream.nix:
```diff
@@ -0,0 +1,129 @@
+  {
+    name = "vaultwarden-litestream";
+    meta = {
+      maintainers = with lib.maintainers; [ ];
```
Please write yourself into here

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-already-reviewed/2617/2919

ak2k committed:
nixos/tests: add vaultwarden-litestream integration test

Integration test for Vaultwarden with Litestream continuous replication.
Validates cross-service file access, WAL-mode backup, disaster recovery
with checksum verification, and service restart.

Uses runTest module style per #386873.

ak2k committed:
nixos/tests/vaultwarden-litestream: use tmpfiles.settings, drop redundant interpolation
